pkg/controller/operator/seed/resources/nodeportproxy/deployments.go

```diff
@@ -189,6 +196,31 @@ func EnvoyDeploymentCreator(cfg *operatorv1alpha1.KubermaticConfiguration, seed
 	}
 }
 
+func HostnameAntiAffinity(app string) *corev1.Affinity {
+	return &corev1.Affinity{
+		PodAntiAffinity: &corev1.PodAntiAffinity{
+			PreferredDuringSchedulingIgnoredDuringExecution: hostnameAntiAffinity(app),
+		},
+	}
+}
+
+func hostnameAntiAffinity(app string) []corev1.WeightedPodAffinityTerm {
+	return []corev1.WeightedPodAffinityTerm{
+		// Avoid that we schedule multiple same-kind pods of a cluster on a single node
+		{
+			Weight: 10,
+			PodAffinityTerm: corev1.PodAffinityTerm{
+				LabelSelector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						resources.AppLabelKey: app,
+					},
+				},
+				TopologyKey: resources.TopologyKeyHostname,
+			},
+		},
+	}
+}
```

Comment:
Why is this function different than the original one (https://github.com/kubermatic/kubermatic/blob/master/pkg/resources/antiaffinity.go)?

Reply:
cause we don't have
'''
ClusterLabelKey: clusterName,
'''
here